src/bytecode/instructions.rs

```diff
@@ -11,6 +11,40 @@ pub enum Value {
     Str(String),
 }
 
+impl Value {
+
+    /// Check whether the Value can be converted into a Float.
+    pub fn is_float(&self) -> bool {
+        match self {
+            Value::Float(_) | Value::Str(_) => true,
+            _ => false
+        }
+    }
+
+    /// Convert the Value to an f64. This conversion returns Some when the value is
+    /// either an Integer, a Float or a Str.
+    /// In Lua, string are always converted to floats when they are used in
+    /// arithmetic expressions.
+    pub fn to_float(&self) -> Option<f64> {
+        match self {
+            Value::Integer(value) => Some(*value as f64),
+            Value::Float(value) => Some(*value),
+            Value::Str(value) => value.parse().ok(),
+            _ => None
+        }
+    }
+
+    /// Convert the Value to an i64. This conversion returns Some when the value is
+    /// either an Integer, or a Float.
+    pub fn to_int(&self) -> Option<i64> {
+        match self {
+            Value::Integer(value) => Some(*value),
+            Value::Float(value) => Some(*value as i64),
```

Comment:
Unfortunately `as` conversions from floats to ints need additional special handling https://doc.rust-lang.org/reference/expressions/operator-expr.html#type-cast-expressions:

> Casting from a float to an integer will round the float towards zero
>
>    NOTE: currently this will cause Undefined Behavior if the rounded value cannot be represented by the target integer type. This includes Inf and NaN. This is a bug and will be fixed.

Reply:
Actually I don't need this conversion. I realized that Lua never converts floats to integers. Also, thank you for pointing this out! I never knew that casting floats to ints might cause undefined behaviour.